src/main/java/com/monumental/services/MonumentService.java

```diff
@@ -129,20 +133,6 @@ private void buildDWithinQuery(CriteriaBuilder builder, CriteriaQuery query, Roo
      * TODO: Be sure to update this whenever the search function changes so that they stay in sync
      */
     public Integer countSearchResults(String searchQuery, String latitude, String longitude, String distance) {
-
-        CriteriaBuilder builder = this.getCriteriaBuilder();
-        CriteriaQuery<Long> query = builder.createQuery(Long.class);
-        Root<Monument> root = query.from(Monument.class);
-        query.select(builder.countDistinct(root));
-
-        if (searchQuery != null) {
-            this.buildSimilarityQuery(builder, query, root, searchQuery, 0.1, false);
-        }
-
-        if (latitude != null && longitude != null && distance != null) {
-            this.buildDWithinQuery(builder, query, root, latitude, longitude, Integer.parseInt(distance));
-        }
-
-        return this.getEntityManager().createQuery(query).getSingleResult().intValue();
+        return this.search(searchQuery, "1", "25", latitude, longitude, distance).size();
```

Comment:
This is not correct. Size will be the paginated size. `countDistinct` is the size of ALL the results not just those in the page.

Reply:
I refactored how `search()` works so it can handle `null` values for `limit` and `page`.

Reply:
This is still a much slower query to make than a `count()` query. Whichever way we do it, we need to be using `countDistinct` for efficiency

Reply:
Oh my bad, I was confused what `countDistinct` was.

You're saying we should do something similar to: https://stackoverflow.com/questions/2883887/in-jpa-2-using-a-criteriaquery-how-to-count-results right?

Reply:
Yes which is basically what we had except with countDistinct instead of count

Reply:
I see.

I wanted to try to find a way to not have the search logic duplicated.

I'll fix it up and see if I can do that another way.

Reply:
Fixed it up. Should be good to go now.